src/ppx_const.ml

```diff
@@ -1,115 +1,149 @@
-open Ast_mapper
-open Ast_helper
-open Asttypes
-open Parsetree
-open Longident
+open Ppxlib
 
-let const_mapper argv =
-  (* Our const_mapper only overrides the handling of expressions in the default mapper. *)
-  { default_mapper with expr =
+(* Shadow polymorphic equality functions from the standard library to
+   avoid future issues *)
+let (=) : unit -> unit -> bool = (=)
+let (<>) : unit -> unit -> bool = (=)
```

Comment:
Would you mind explaining to me this set of assignments?
Wouldn't assigning (<>) to (=) interfere with the correct operation/recognition of <> ? Maybe I don't understand this syntax.

Reply:
As far as I understand this is basically making the polymorphic comparison operators inaccessible/unworkable, to discourage their usage. Polymorphic comparison can be dangerous, see https://blog.janestreet.com/the-perils-of-polymorphic-compare/

Personally I would have done something like `let (=) = failwith "Disable polymorphic compare"`, to enforce it.

Reply:
Notice that in this PR the explicit comparison functions like `String.equal` and `Int.equal` are being used, instead of the polymorpchic compare.